### PR TITLE
CONCF-497 Curated Content on Main Page

### PR DESCRIPTION
**Pull request by bognix**
ping @rogatty @vforge - curated content on mercury main page

### CODE REVIEW

File: front/scripts/main/models/CuratedContentModel.ts [hunk 1]
```diff
@@ -5,8 +5,10 @@
 App.CuratedContentModel = Em.Object.extend({
 	activeSectionItems: [],
 	cachedSectionItems: {},
+	activeSection: false,
 
 	fetchItemsForSection: function(sectionName: string) : any {
```
Comment (rogatty) [inline front/scripts/main/models/CuratedContentModel.ts]: `: Em.RSVP.Promise`

File: server/lib/MediaWiki.ts [hunk 2]
```diff
@@ -124,7 +124,7 @@ export class ArticleRequest {
 	curatedContentSection (sectionName: string): Promise<any> {
 		var url = createUrl(this.wikiDomain, 'wikia.php', {
 			controller: 'MercuryApi',
-			method: 'getCuratedContentSection',
+			method: 'getItemsForCuratedContentSection',
```
Comment (rogatty) [inline server/lib/MediaWiki.ts]: This is not correct, see https://github.com/Wikia/app/blob/dev/extensions/wikia/MercuryApi/MercuryApiController.class.php#L349

File: front/templates/main/components/main-page.hbs [hunk 3]
```diff
@@ -2,5 +2,5 @@
 	{{featured-content model=model.featuredContent}}
 {{/if}}
 {{#if model.curatedContent}}
-	{{curated-content model=model.curatedContent}}
+	{{curated-content curatedContent=model.curatedContent}}
```
Comment (rogatty) [inline front/templates/main/components/main-page.hbs]: Why not `model`? It is a model after all.

File: front/styles/main/component/_curated-content.scss [hunk 4]
```diff
@@ -1,34 +1,137 @@
...
+		}
+
+		&.grid-icon {
+			display: table-cell;
+			vertical-align: bottom;
+
+			.grid-icon-container {
+				background: rgba(0, 0, 0, .4);
+				-moz-border-radius: 100px 0 0;
+				-webkit-border-radius: 100px 0 0;
```
Comment (vforge) [inline front/styles/main/component/_curated-content.scss]: Browser-specific declarations are obsolete for `border-radius`

File: front/svg/symbols/main/category_thumb_icon.svg [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<svg version="1.0" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
+	 viewBox="0 0 34 34" style="enable-background:new 0 0 34 34;" xml:space="preserve" fill="#FFF">
```
Comment (vforge) [inline front/svg/symbols/main/category_thumb_icon.svg]: please remove this `fill` attribute, it overrides CSS styles. You can also remove `<?xml` header and `<g>` group declaration.

File: front/styles/main/component/_curated-content.scss (hunk 4 above)
Comment (rogatty) [inline front/styles/main/component/_curated-content.scss]: We're using https://www.npmjs.com/package/gulp-autoprefixer anyway, no need to add prefixes manually.

File: front/templates/main/components/main-page.hbs (hunk 3 above)
Comment (bognix) [inline front/templates/main/components/main-page.hbs]: this piece of code would be refactored as part of implementing third level of nesting.

File: front/scripts/main/mixins/ViewportMixin.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+		height: null,
+		width: null
+	},
+	initiated: false,
+
+	init: function (): void {
+		this._super();
+		if (!this.get('initiated')) {
+			this.set('viewportDimensions.width', Math.max(document.documentElement.clientWidth, window.innerWidth || 0));
+			this.set('viewportDimensions.height', Math.max(document.documentElement.clientHeight, window.innerHeight || 0));
```
Comment (rogatty) [inline front/scripts/main/mixins/ViewportMixin.ts]: How about replacing these two lines with `this.onResize()`?

File: front/templates/main/components/curated-content-item.hbs [hunk 7]
```diff
@@ -1,5 +1,18 @@
...
+	{{#if imageUrl}}
+		{{thumbnail imageUrl width=imageSize height=imageSize alt=label mode=cropMode}}
+	{{/if}}
+	</div>
+	<figcaption class='item-figcaption'>
+		<div class='item-caption-row'>
+		<span class='item-caption label'>{{label}}</span>
+		<span class='item-caption grid-icon'>
+			<div class='grid-icon-container'>
+				{{svg 'category_thumb_icon' role='img' class='category-icon' viewBox='0 0 17 17'}}
```
Comment (rogatty) [inline front/templates/main/components/curated-content-item.hbs]: Make it dasherized like others (`category-thumb-icon`).

File: front/templates/main/components/curated-content.hbs [hunk 8]
```diff
@@ -1,20 +1,23 @@
...
+			<div class='section-title'>{{model.activeSection}}</div>
+		</div>
+		{{#each item in model.activeSectionItems}}
+			<a {{action 'trackClick' 'curated-content' 'item-click'}} href='{{item.article_local_url}}'>
+				{{curated-content-item label=item.label imageUrl=item.image_url}}
+			</a>
+		{{/each}}
+	</nav>
 {{/if}}
+{{loading-spinner active=isLoading overlay=true}}
```
Comment (rogatty) [inline front/templates/main/components/curated-content.hbs]: `overlay` is true by default.

File: front/scripts/main/mixins/ViewportMixin.ts (hunk 6 above)
Comment (bognix) [inline front/scripts/main/mixins/ViewportMixin.ts]: I would fix it inside https://github.com/Wikia/mercury/pull/763

**Comment by rogatty** [comment]
LGTM

**Pull request merged**